tests/scalers/neo4j/neo4j_test.go

```diff
@@ -0,0 +1,296 @@
+	       (ac9:Person { name: "Soha", from: "Bhopal", profession: "Charter Accountant" }),
+	       (ac10:Person { name: "Ananya", from: "Paris", popularfor: "Lawn Tennis" }),
+	       (ac11:Person { name: "Anna", from: "Moscow", profession: "Software Engineer" }),
+	       (ac1)-[:FOLLOWS]->(ac2),(ac2)<-[:FOLLOWS]-(ac1),(ac5)-[:FOLLOWS]->(ac1),
+	       (ac6)-[:FOLLOWS]->(ac1),(ac8)-[:FOLLOWS]->(ac1),(ac9)-[:FOLLOWS]->(ac1),
+	       (ac11)-[:FOLLOWS]->(ac1),(ac5)-[:FOLLOWS]->(ac2),(ac6)-[:FOLLOWS]->(ac2),
+	       (ac5)-[:FOLLOWS]->(ac3),(ac6)-[:FOLLOWS]->(ac3),(ac8)-[:FOLLOWS]->(ac3),
+	       (ac9)-[:FOLLOWS]->(ac3),(ac11)-[:FOLLOWS]->(ac3),(ac5)-[:FOLLOWS]->(ac4),
+	       (ac6)-[:FOLLOWS]->(ac4),(ac8)-[:FOLLOWS]->(ac4),(ac9)-[:FOLLOWS]->(ac4),
+	       (ac11)-[:FOLLOWS]->(ac4),(ac5)-[:FOLLOWS]->(ac7),(ac6)-[:FOLLOWS]->(ac7),
+	       (ac5)-[:FOLLOWS]->(ac10),(ac6)-[:FOLLOWS]->(ac10),(ac8)-[:FOLLOWS]->(ac10),
+	       (ac5)<-[:FOLLOWS]-(ac6),(ac5)<-[:FOLLOWS]-(ac8),(ac5)<-[:FOLLOWS]-(ac11),
+	       (ac6)<-[:FOLLOWS]-(ac5),(ac6)<-[:FOLLOWS]-(ac8),(ac6)<-[:FOLLOWS]-(ac11),
+	       (ac6)<-[:FOLLOWS]-(ac9),(ac11)<-[:FOLLOWS]-(ac9),(ac11)<-[:FOLLOWS]-(ac6),
+	       (ac8)<-[:FOLLOWS]-(ac11),(ac8)<-[:FOLLOWS]-(ac9),(ac9)<-[:FOLLOWS]-(ac5),
+	       (ac10)<-[:FOLLOWS]-(ac7),(ac4)<-[:FOLLOWS]-(ac7),(ac3)<-[:FOLLOWS]-(ac7),
+	       (ac3)<-[:FOLLOWS]-(ac10),(ac1)<-[:FOLLOWS]-(ac7),(ac2)<-[:FOLLOWS]-(ac7),
+	       (ac3)<-[:FOLLOWS]-(ac7),(ac3)<-[:FOLLOWS]-(ac1),
+	       (ac10)<-[:FOLLOWS]-(ac3),(ac10)<-[:FOLLOWS]-(ac2),(ac10)<-[:FOLLOWS]-(ac1)
+	       return ac1,ac2,ac3,ac4,ac5,ac6,ac7,ac8,ac9,ac10,ac11',{{.UsernameArg}},{{.PasswordArg}}]
+  restartPolicy: OnFailure
+`
+	KubectlApplyWithTemplate(t, data, "activationPodTemplate", activationPodTemplate)
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation ---")
+	deployPodActivation(t, data)
+	time.Sleep(time.Second * 60)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 1),
+		"replica count should be %d after 1 minute", minReplicaCount)
+}
+
+func deployPodUp(t *testing.T, data templateData) {
+	const scaleUpPodTemplate = `apiVersion: v1
+kind: Pod
+metadata:
+  name: neo4j-demo-up
+  namespace: {{.TestNamespace}}
+spec:
+  containers:
+  - image: tanishabanik/neo4j-demo:0.0.8
+    name: neo4j-demo-up
+    args: ["neo4j://{{.ScalerName}}.{{.TestNamespace}}.svc.cluster.local:7687",
+	      "match(e:Person{name:'Robert'}),(d:Person{name:'Saurav'}) create (e)-[m:FOLLOWS]->(d) return e,m,d",{{.UsernameArg}},{{.PasswordArg}}]
+  restartPolicy: OnFailure
+`
+	KubectlApplyWithTemplate(t, data, "scaleUpPodTemplate", scaleUpPodTemplate)
+}
+
+func testScaleOut(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing scale out ---")
+	deployPodUp(t, data)
+	time.Sleep(time.Second * 60)
```

Comment:
This is already included in the `WaitForDeploymentReplicaReadyCount`/`AssertReplicaCountNotChangeDuringTimePeriod` functions, please remove it

Reply:
I am removing the `time.Sleep` line